src/metadata.c

```diff
@@ -102,9 +131,15 @@ int MetadataRead(Metadata *m, const char *filename)
     RedisModule_Assert(elem_count == METADATA_ELEM_COUNT);
     RedisModule_Assert(strncmp(str, METADATA_STR, strlen(METADATA_STR)) == 0);
     RedisModule_Assert(version == METADATA_VERSION);
+    RedisModule_Assert(strlen(dbid) == RAFT_DBID_LEN);
 
+    strcpy(m->dbid, dbid);
```

Comment:
why not to expand `MetadataConfigure()` to all metadata fields configuration, and call `MetadataConfigure()` here?